var/spack/repos/builtin/packages/aml/package.py

```diff
@@ -9,60 +9,122 @@
 
 
 class Aml(AutotoolsPackage):
-    """AML: Building Blocks for Memory Management."""
+    """AML is a memory management library designed to ease the use of complex
+    memory topologies and complex data layout optimizations for
+    high-performance computing applications.
+    """
 
     homepage = "https://argo-aml.readthedocs.io/"
-    url = "https://www.mcs.anl.gov/research/projects/argo/downloads/aml-0.1.0.tar.gz"
-    git = "https://github.com/anlsys/aml.git"
+
     maintainers = ['perarnau']
 
     test_requires_compiler = True
 
     tags = ['e4s']
 
-    version('0.1.0', sha256='cc89a8768693f1f11539378b21cdca9f0ce3fc5cb564f9b3e4154a051dcea69b')
-    version('master', branch='master', submodules=True)
-
+    # Package sources
+    ###################################
+
+    url = 'https://github.com/anlsys/aml/releases/download/v0.2.0/aml-0.2.0.tar.gz'
+    git = 'https://github.com/anlsys/aml.git'
+
+    # version string is generated from git tags, requires entire repo
+    version('master', branch='master', submodules=True, get_full_repo=True)
+
+    version('0.2.0',
+            sha256='2044a2f3f1d7a19827dd9c0726172b690189b4d3fe938656c4160c022468cc4a')
+    version('0.1.0',
+            sha256='cc89a8768693f1f11539378b21cdca9f0ce3fc5cb564f9b3e4154a051dcea69b',
+            deprecated=True)
+
+    # Generate possible variants.
+    #############################
+
+    variant('opencl', default=False,
+            description="Support for memory operations on top of OpenCL.")
+    variant('ze', default=False,
+            description="Support for memory operations on top of Level Zero.")
+    variant('hip', default=False,
+            description="Support for memory operations on top of HIP.")
+    variant('cuda', default=False,
+            description="Support for memory operations on top of CUDA.")
+    variant('hwloc', default=False,
+            description="Enable feature related to topology management")
+    variant('openmp-targets', values=disjoint_sets(('spir64',)),
+            description="Intel OpenMP target architecture setting.")
+    variant('hip-platform', values=disjoint_sets(('amd', 'nvidia')),
+            description="HIP backend platform.")
```

Comment:
Why are we using `disjoint_sets` here, and declaring only one set? If there is a single set, a multi-valued variant would fit better.[^1]

[^1]: To clarify the comment,`disjoint_set` allows to select multiple values from the same set, but forbids having values from different sets.

Reply:
I removed it as a variant and just enabled it when the compiler is `%oneapi`.